src/test_l1_hdmi_cec_driver.c

```diff
@@ -2231,16 +2238,33 @@ static UT_test_suite_t *pSuite_panel = NULL;
  */
 int test_hdmicec_hal_l1_register( void )
```

Comment:
you should create another function it will be much clearer.

`test_hdmidec_hal_l1_register_sink_tests()`
`test_hdmidec_hal_l1_register_source_tests()`

Call them both..

But inside the function make a decision on your flag, and then return from the function is required.

Reply:
Updated logic.